src/lib.rs

```diff
@@ -243,60 +333,109 @@ fn compile(
+                    .map_err(|e| format!("cannot read from \"{}\": {}", path_lit, e.to_string()))?;
+                let incl = ResolvedInclude { resolved_name: path_lit, content };
+                Ok(incl)
+            });
+            for (k, v) in cfg.defs.iter() {
+                opt.add_macro_definition(&k, v.as_ref().map(|x| x.as_ref()));
+            }
+            if cfg.debug {
+                opt.set_generate_debug_info();
+            }
+
+            let mut compiler = Compiler::new().unwrap();
+            let path = if let Some(path) = path {
+                dep_paths.borrow_mut().push(path.to_owned());
+                path
+            } else { "<inline>" };
+            let out = compiler
+                .compile_into_spirv(src, cfg.kind, &path, &cfg.entry, Some(&opt))
+                .map_err(|e| e.to_string())?;
+            if out.get_num_warnings() != 0 {
+                return Err(out.get_warning_messages());
+            }
+            let spv = out.as_binary().into();
+            let feedback = CompilationFeedback { spv, dep_paths: dep_paths.into_inner() };
+            Ok(feedback)
+        }
+
+        #[cfg(feature = "naga")]
+        InputSourceLanguage::WGSL => {
+            // silence the compiler warning about path not being used when only wgsl is enabled
+            let _ = path.is_some();
+
+            match wgsl::parse_str(&src) {
+                Ok(module) => {
+                    // Attempt to validate WGSL, error if invalid
+                    match Validator::new(ValidationFlags::all(), cfg.capabilities).validate(&module) {
+                        Ok(info) => {
+                            let mut options = spv::Options::default();
+                            if cfg.debug {
+                                options.flags.insert(WriterFlags::DEBUG);
+                            } else {
+                                options.flags.remove(WriterFlags::DEBUG);
+                            }
+                            if cfg.adjust_coordinate_space {
+                                options.flags.insert(WriterFlags::ADJUST_COORDINATE_SPACE);
+                            } else {
+                                options.flags.remove(WriterFlags::ADJUST_COORDINATE_SPACE);
+                            }
+                            options.lang_version = cfg.naga_spirv_version;
+
+                            match spv::write_vec(&module, &info, &options) {
+                                Ok(spv) => {
+                                    let feedback = CompilationFeedback { spv, dep_paths: Vec::new() };
+                                    Ok(feedback)
+                                }
+                                Err(e) => Err(format!("{:?}", e))
+                            }
+                        },
+                        Err(e) => Err(format!("{:?}", e))
+                    }
+                },
+                Err(e) => {
+                    e.emit_to_stderr(src);
```

Comment:
Could you attach the source before the error messages emitted next line? The users usually want more control over the output streams.

Reply:
this should be fixed in the most recent push